### PR TITLE
chore: update dependencies for react 18

### PR DESCRIPTION
**Pull request by noobinthisgame**
## 📝 Description

Bumping all dependency versions to match React 18. Thanks to the brain @TimKolberger who thought about doing so :D

## 🚀 New behavior

Major bumps to React, Chakra and contentlayer. Smaller bumps for other dependencies.

## 💣 Is this a breaking change (Yes/No):

Yes

## 📝 Additional Information

Open todos:
- [x] Fix component overview error with hydration -> edit: seems to onyl be happening on the dev server we can check this later.
- [x] Update component types that have currently FC
- [x] Check all pages and examples
- [x] Fix contentlayer build errors (local build is doing fine)

## Next steps
- to make it less painful next time we should check some of our dependencies I didn't bump and bump them. Especially `"@docsearch/react": "^1.0.0-alpha.27"` latest beeing 3.0.0
- See if we can simplify code base and reduce dependencies
- remove babel ? There is no superjson support right now for swc but do we really need it

### CODE REVIEW

File: src/components/tutorial/tutorial-code-block.tsx [hunk 1]
```diff
@@ -70,6 +70,8 @@ const CodeBlock = ({ path, code, showLineNumbers = false }: CodeBlockProps) => {
 
 export const TutorialCodeBlock = (props: CodeBlockProps) => {
   return (
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore - sanpack has not yet fully provided React 18 support
```
Comment (TimKolberger) [inline src/components/tutorial/tutorial-code-block.tsx:74]:
```suggestion
    // @ts-ignore - sandpack has not yet fully provided React 18 support
```

**Review by TimKolberger — COMMENTED** [review]
I found a few typos and some leftover `ReactDOM.render` calls in the docs:

image

Kindly replace those with

```tsx
import * as ReactDOM from "react-dom/client"
const rootElement = document.getElementById("root")
ReactDOM.createRoot(rootElement).render(<COMP />)
```

**Comment by noobinthisgame** [comment]
@TimKolberger I've added the changes :) Let me know if you think we are good to go here.

**Review by TimKolberger — APPROVED** [review]
Awesome 👏 thanks @noobinthisgame!
🚀

**Pull request merged**